unity-client/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarModifiers/AvatarModifierArea.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using DCL;
+using DCL.Components;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class AvatarModifierArea : BaseComponent
+{
+    [Serializable]
+    public class Model
+    {
+        // TODO: Change to TriggerArea and handle deserialization with subclasses
+        public BoxTriggerArea area;
+        public string[] modifiers;
+
+    }
+
+    [HideInInspector]
+    public Model model = new Model();
+
+    private event Action<GameObject> OnAvatarEnter;
+    private event Action<GameObject> OnAvatarExit;
+    private readonly Dictionary<string, AvatarModifier> modifiers;
+
+    public AvatarModifierArea()
+    {
+        // Configure all available modifiers
+        this.modifiers = new Dictionary<string, AvatarModifier>()
+        {
+            { "HIDE_AVATARS", new HideAvatarsModifier() },
+            { "DISABLE_PASSPORTS", new DisablePassportModifier() }
+        };
+    }
+
+    public override IEnumerator ApplyChanges(string newJson)
+    {
+
+        // Clean previous listeners
+        OnAvatarEnter = null;
+        OnAvatarExit = null;
+
+        model = SceneController.i.SafeFromJson<Model>(newJson);
+        if (model.modifiers != null)
+        {
+            model.area.AddCollider(gameObject);
+
+            // Add all listeners
+            foreach (string modifierKey in model.modifiers)
+            {
+                AvatarModifier modifier = this.modifiers[modifierKey];
+                OnAvatarEnter += modifier.ApplyModifier;
+                OnAvatarExit += modifier.RemoveModifier;
+            }
+        }
+
+        yield return null;
+    }
+
+    private void OnDestroy()
+    {
+        Collider collider = model?.area?.GetCollider(gameObject);
+        if (collider != null)
+        {
+            Destroy(collider);
+        }
```

Comment:
The null check is not needed, if you pass `null` to `Object.Destroy` it will behave just fine.

Reply:
Done :)